src/ConstructiveSolidGeometry/VolumePrimitives/Cone.jl

```diff
@@ -1,207 +1,850 @@
-struct Cone{T,TR,TP,TZ} <: AbstractVolumePrimitive{T}
-    r::TR #if not a Tuple, then Cone is a Tube
-    φ::TP
-    z::TZ
-    #if r is a Tuple, the first entry refers to the r-interval at the bottom, the second one to the r-interval at the top
-    function Cone( ::Type{T},
-                   r::Union{T, <:AbstractInterval{T}, Tuple{T,T}, Tuple{I,I}},
-                   φ::Union{Nothing, <:AbstractInterval{T}},
-                   z::Union{T, <:AbstractInterval{T}}) where {T, I<:AbstractInterval{T}}
-        new{T,typeof(r),typeof(φ),typeof(z)}(r, φ, z)
+"""
+    struct Cone{T,CO,RT,TP} <: AbstractVolumePrimitive{T, CO}
+
+T: Type of values, e.g. Float64
+CO: ClosedPrimitive or OpenPrimitive <-> whether surface belongs to it or not
+
+* `r::TR`: 
+    * TR = Real -> Cylinder
+    * TR = (Real, Real) -> Tube (r_in = r[1], r_out = r[2])
+    * TR = ((Real,), (Real,)) Solid widening Cylinder  -> (r_bot = r[1][1], r_top = r[1][2])
+    * TR = ((Real,Real), (Real,Real)) Solid widening Tube ->\n(r_bot_in = r[1][1], r_bot_out = r[1][2], r_top_in = r[2][1], r_top_out = r[2][2])
+    * TR = (Nothing, (Real,Real)) Cone ->\n(r_bot_in = r_bot_out = 0, r_top_in = r[2][1], r_top_out = r[2][2])
+    * TR = ((Real,Real), Nothing) Cone ->\n(r_bot_in = r[1][1], r_bot_out = r[1][2], r_top_in = r_top_out = 0)
+    * ... (+ elliptical cases -> (a, b))
+    * Not all are implemented yet
+
+* `φ::TP`: 
+    * TP = Nothing <-> Full in φ
+    * ...
+* `zH::T`: half hight/length of the cone
+"""
+@with_kw struct Cone{T,CO,RT,TP} <: AbstractVolumePrimitive{T, CO}
+    r::RT = 1
+    φ::TP = nothing
+    hZ::T = 1
+
+    origin::CartesianPoint{T} = zero(CartesianPoint{T})
+    rotation::SMatrix{3,3,T,9} = one(SMatrix{3, 3, T, 9})
+end
+
+Cone{T,CO,RT,TP}( c::Cone{T,CO,RT,TP}; COT = CO,
+            origin::CartesianPoint{T} = c.origin,
+            rotation::SMatrix{3,3,T,9} = c.rotation) where {T,CO<:Union{ClosedPrimitive, OpenPrimitive},RT,TP} =
+    Cone{T,COT,RT,TP}(c.r, c.φ, c.hZ, origin, rotation)
+
+####################################################################
+####################################################################
+
+
+#=
+___
+   |
+   |
+   |
+___|
+
+=#
+const Cylinder{T,CO} = Cone{T,CO,T,Nothing} # Full in φ
```

Comment:
So, no more `Tube`?
We call it `tube` in the config files, shouldn't the code be consistent with that?

Reply:
To me a tube has a borehole. That is why I changed it in the code. 
Hmm, the config file does not really depend on it. Both, `Tube` and `Cone` just 
lead to the constructor of a Cone. 
But, yes, we should clean this up.
I also have some other (minor) comments in my head regarding the config files. 
I would suggest to clean this up after this PR.